nix/flake/flakeref.go

```diff
@@ -396,7 +397,13 @@ func splitPathOrOpaque(u *url.URL) ([]string, error) {
 	upath = path.Clean(upath)
 
 	var err error
-	split := strings.Split(upath, "/")
+	var split []string
+	if allowSlashesIfRef {
+		split = strings.SplitN(upath, "/", 3)
```

Comment:
I don't understand what scenario is being dealt with here. What does `allowSlashesifRef` do, and why does it only care about 3 slashes, while the `else` can deal with an unlimited number of slashes?

Reply:
I think this is to fix a bug where we weren't parsing github refs with slashes in them correctly. I broke out the flakeref fixes into a separate PR just so they can be tracked separately - https://github.com/jetpack-io/devbox/pull/1863.